packages/material-ui-styles/src/styled/styled.d.ts

```diff
@@ -9,8 +9,8 @@ import * as React from 'react';
 /**
  * @internal
  */
-export type ComponentCreator<C extends React.ElementType> = <Theme>(
-  styles: CSSProperties | ((theme: Theme) => CSSProperties),
+export type ComponentCreator<C extends React.ElementType> = <Theme, Props extends {} = any>(
+  styles: CSSProperties | (({ theme, ...props }: { theme: Theme; props: Props }) => CSSProperties),
```

Comment:
This suggests that `props` has the type `{ props: Props }`. I think it should have to type `Props` at which point
```suggestion
  styles: CSSProperties | (({ theme, ...props }: { theme: Theme } & Props) => CSSProperties),
```
should be correct.

Reply:
Done.